cmd/observer/node_builder/observer_builder.go

```diff
@@ -994,7 +992,33 @@ func (builder *ObserverServiceBuilder) enqueueRPCServer() {
 		if err != nil {
 			return nil, err
 		}
-		engineBuilder.WithRouting(proxy)
+
+		/*
+			TODO:
+			The following code to setup the connection to the access node is temporary
+			The forwarding should allow for multiple upstream access nodes, allow for retries, and track metrics
+		*/
+		if len(builder.upstreamIdentities) == 0 {
+			panic("please specify an upstream identity")
+		}
+
+		identity := builder.upstreamIdentities[0]
+
+		tlsConfig, err := grpcutils.DefaultClientTLSConfig(identity.NetworkPubKey)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get default TLS client config using public flow networking key %s %w", identity.NetworkPubKey.String(), err)
+		}
+
+		conn, _ := grpc.Dial(
+			identity.Address,
+			grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(grpcutils.DefaultMaxMsgSize)),
+			grpc.WithTransportCredentials(credentials.NewTLS(tlsConfig)),
+			backend.WithClientUnaryInterceptor(time.Minute))
+
+		client := accessproto.NewAccessAPIClient(conn)
```

Comment:
Is it ok to do anything fancy right now when connecting to the upstream access node? I could add something like looping over all upstream-node-addresses and picking the first one that works. There was some more robust code in /apiproxy directory I removed. But I could add some of the features back

Reply:
you can add in the minimum feature of being able to connect to multiple staked ANs, and the retry strategy you described above. No need for the round robin lookup though, you can always loop over them in the order they were specified in the args

Reply:
Implemented